### PR TITLE
31 create test for select function

### PR DESCRIPTION
**Pull request by zth96**
Test select function with a single column
Test select function with multiple columns
Test select function with a non-existing column
Test select function with no columns specified

### CODE REVIEW

**Review by farrandi — COMMENTED** [review]
Look at Thomas' PR to use pytest.fixture #33 

I would suggest using that, maybe wait for him to merge first

File: tests/test_tidyversetopandas.py [hunk 1]
```diff
@@ -1,23 +1,100 @@
...
+
+
+def test_select_no_columns():
+    """Test select function with no columns specified"""
+    df = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
+    result = tidyversetopandas.select(df)
+    assert result.empty
+    assert len(result.columns) == 0
+    
+=======
```
Comment (786213750) [inline tests/test_tidyversetopandas.py]: remove 
```suggestion

```

File: tests/test_tidyversetopandas.py [hunk 2]
```diff
@@ -1,23 +1,100 @@
...
+    """Test filter function that raise error for type dataframe"""
+    with pytest.raises(TypeError):
+        _ = tidyversetopandas.filter("abc", "A > 1")
+
+
+def test_filter_query_error(input_df_2):
+    """Test filter function that raise error for type query"""
+    with pytest.raises(TypeError):
+        _ = tidyversetopandas.filter(input_df_2, pd.DataFrame([]))
+>>>>>>> main
```
Comment (786213750) [inline tests/test_tidyversetopandas.py]: remove
```suggestion

```

File: tests/test_tidyversetopandas.py [hunk 3]
```diff
@@ -1,23 +1,100 @@
...
-    df = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
-    df = tidyversetopandas.mutate(df, c=lambda x: x["a"] + x["b"])
+    df = tidyversetopandas.mutate(input_df_1, c=lambda x: x["a"] + x["b"])
     assert df["c"].tolist() == [5, 7, 9]
+
+
+<<<<<<< HEAD
+def test_select_single_column():
+    """Test select function with a single column"""
+    df = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
```
Comment (786213750) [inline tests/test_tidyversetopandas.py]: you can use fixture method for this
```suggestion
    input_df_1
```

File: tests/test_tidyversetopandas.py [hunk 4]
```diff
@@ -1,23 +1,100 @@
...
+    result = tidyversetopandas.select(df, "a")
```
Comment (786213750) [inline tests/test_tidyversetopandas.py]:
```suggestion
    result = tidyversetopandas.select(input_df_1, "a")
```

**Comment by zth96** [comment]
Thanks guys. I updated the test to include pytest.fixture. Please check and let me know.

File: tests/test_tidyversetopandas.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
 from tidyversetopandas import tidyversetopandas
 import pandas as pd
-
+import pytest
```
[the remark this answers is not in the document]
Comment (zth96) [reply tests/test_tidyversetopandas.py:3]: I will remove it.

**Review by LilyTao0531 — APPROVED** [review]

**Comment by LilyTao0531** [comment]
Looks good!

**Review by LilyTao0531 — APPROVED** [review]

**Review by 786213750 — COMMENTED** [review]
looks good!

**Review by 786213750 — APPROVED** [review]

**Comment by LilyTao0531** [comment]
Looks good!

**Pull request merged**